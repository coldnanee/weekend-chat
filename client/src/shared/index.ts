export { SocketProvider, useSocketContext } from "./ui/socket/";
export { ReactQueryProvider } from "./lib/react-query";

export { DefaultAvatar } from "./ui/default-avatar";

export { Loader } from "./ui/loader";
export { createMetadata } from "./lib/createMetadata";
export { default } from "./lib/axios";
