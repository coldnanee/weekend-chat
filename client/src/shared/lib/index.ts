export { ReactQueryProvider } from "./react-query";
export { createMetadata } from "./createMetadata";
export { default } from "./axios";
export { getFormattedIsoDate } from "./getFormattedIsoDate";
export { refreshToken } from "./refreshToken";
export { i18n } from "./i18n";
