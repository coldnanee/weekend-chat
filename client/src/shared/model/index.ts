export { useSocketStore } from "./store";
export { refreshToken } from "./refreshToken";
