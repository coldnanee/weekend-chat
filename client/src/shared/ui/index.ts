export { Loader } from "./loader";
export { DefaultAvatar } from "./default-avatar";
export { SocketProvider } from "./socket";
export { SocketHandlers } from "./socket-handlers";
