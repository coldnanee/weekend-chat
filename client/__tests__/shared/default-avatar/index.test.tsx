import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { DefaultAvatar } from "@/shared";

describe("DefaultAvatar", () => {
	test("render", () => {
		render(
			<DefaultAvatar
				alt="image"
				width={40}
				height={40}
			/>
		);
		expect(screen.getByTestId("default-avatar")).toBeInTheDocument();
		expect(screen.getByTestId("default-avatar")).toMatchSnapshot();
	});

	test("check image default", () => {
		render(
			<DefaultAvatar
				width={50}
				height={50}
				alt="default"
			/>
		);
		expect(
			screen.getByTestId("default-avatar").getAttribute("src")
		).not.toBeNull();
	});

	test("check correct props", () => {
		render(
			<DefaultAvatar
				width={25}
				height={25}
				alt="default"
				className="RANDOM_CLASS"
				src="http://RANDOM_SRC.jpg"
			/>
		);
		expect(
			screen.getByTestId("default-avatar").getAttribute("src")
		).not.toBeNull();
		expect(screen.getByTestId("default-avatar")).toHaveClass("RANDOM_CLASS");
	});
});
