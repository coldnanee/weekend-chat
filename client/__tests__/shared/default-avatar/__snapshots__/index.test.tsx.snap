// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultAvatar render 1`] = `
<img
  alt="image"
  class=" root"
  data-nimg="1"
  data-testid="default-avatar"
  decoding="async"
  fetchpriority="high"
  height="40"
  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
  style="color: transparent; width: 40px; height: 40px;"
  width="40"
/>
`;
